client/src/components/app.jsx

```diff
@@ -88,12 +107,19 @@ class App extends React.Component {
             clickFunction={this.previousSlide}
             glyph="&#9664;" />
 
+          <div className="exit-button" onClick={() => this.handleExitClick()}><span>X</span></div>
+
           <ImageSlide url={this.state.photos[this.state.currentImageIndex].url} />
 
           <Arrow
             direction="right"
             clickFunction={this.nextSlide}
             glyph="&#9654;" />
+
+          <Footer
+            publisher={this.state.photos[this.state.currentImageIndex].publisherName}
```

Comment:
Destructuring at the top of the render function would make this easier to read.. you can do ``` const { photos, currentImageIndex } = this.state ``` and then use those variables directly :+1: